_P221_LandisGyrUH550.ino

```diff
@@ -0,0 +1,450 @@
+#define USES_P221
+#ifdef USES_P221
+//#######################################################################################################
+//############################# Plugin 218: read Landis & Gyr UH550 #####################################
+//#######################################################################################################
+
+//IR RX/TX sensor based on https://wiki.volkszaehler.org/hardware/controllers/ir-schreib-lesekopf
+
+//Device pin 1 = RX
+//Device pin 2 = TX
+
+#include <ESPeasySoftwareSerial.h>
+#define PLUGIN_221
+#define PLUGIN_ID_221 221
+#define PLUGIN_NAME_221 "Communication - Landis & Gyr UH550 [TESTING]"
+#define PLUGIN_VALUENAME1_221 "Heat"
+#define PLUGIN_221_VARIABLECOUNT 1
+
+// Search filters in received data.
+// NOTE: This data remains in Flash!
+const char initFilter[] PROGMEM = {"/LUGCUH50"};
+const char qh_Filter[] PROGMEM = {"6.8("};                // 6.8 - Quantity of Heat (GJ)
+const char vol_Filter[] PROGMEM = {"6.26("};              // 6.26 - Volume (m3)
+const char qhPrev_Filter[] PROGMEM = {"6.8*01("};         // 6.8*01 - Quantity of Heat previous year (GJ)
+const char volPrev_Filter[] PROGMEM = {"6.26*01("};       // 6.26*01 - Volume previous year (m3)
+const char poMax_Filter[] PROGMEM = {"6.6("};             // 6.6 - Power Max (kW)
+const char inOutFlowMaxTemp_Filter[] PROGMEM = {"9.4("};  // 9.4 - max inflow & max reflow temperature (C°)
+
+const char* const filters[] PROGMEM = { qh_Filter, vol_Filter, inOutFlowMaxTemp_Filter, qhPrev_Filter, volPrev_Filter, poMax_Filter }; 
+const byte numberFilters = sizeof(filters) / sizeof(filters[0]);
+
+boolean Plugin_221_init = false;
+byte PIN_SER_RX = 0;
+byte PIN_SER_TX = 0;
+
+struct ValueUnitPair
+{
+  char value[10];
+  char unit[10];
+};
+
+struct DataSet
+{
+  char address[10];
+  byte vupCount;
+  struct ValueUnitPair vup[20];
+};
+
+struct DataSet datasets[10];
+byte indexDS = 0;
+bool messageProcessingComplete = false;
+
+boolean Plugin_221(byte function, struct EventStruct *event, String &string)
+{
+  boolean success = false;
+
+  Serial.begin(115200);

```

Comment:
Why always send logs to Serial and not to the normal log using `addLog` ?
This will affect other plugins using the serial port, since this is always being called as soon as this plugin is enabled or loaded in the web page.
Also simply fetching the names of the variables of this plugin will reset the Serial port to this baud rate. (then the plugin doesn't have to be enabled)

Reply:
serial logger will be avoided soon